src/ray/rpc/client_call.h

```diff
@@ -229,6 +229,9 @@ class ClientCallManager {
     // pointer.
     auto tag = new ClientCallTag(call);
     call->response_reader_->Finish(&call->reply_, &call->status_, (void *)tag);
+    // Make a mock post here in order to record the number of requests that are replied.
+    // .request - .reply_received == number of in flight grpc requests.
+    main_service_.post([]() {}, tag->GetCall()->GetName() + ".request");
```

Comment:
This is a bit sketchy since it might get delayed anyways. Why not "start" the request here, and then wrap the ClientCallback here to record the "end"? We can add manual start/end support to the instrumentation.

Reply:
Agreed on the sketchiness, and that manual start/end instrumentation hooks would be better here than posting a dummy handler to the event loop. Two things though:

1. I have a P0 in my sprint to add manual instrumentation hooks in order to instrument deadline timers and local stream sockets, and I have an idea of how I'm going to refactor the event loop instrumentation to make it not hacky, so I'd vote for holding off on adding that and either keeping what you currently have or removing it.
2. IMO, tracking the number of active gRPC requests is out-of-scope for instrumentation of the asio event loop, and should be done at the gRPC instrumentation level, either manually using gRPC interceptors or automatically using the OpenCensus gRPC plugin. I was planning on exploring this as part of a larger refactor of the event loop instrumentation to make the stats collection be more general (e.g. with manual start/end hooks).

Reply:
Re @ericl Can you tell me what it means by it gets delayed anyway? 

Re @clarkzinzow 
The current implementation is the easiest solution, and while I agree this is better being in the grpc layer, we need this feature for memory debugging. What about we just keep the current implementation and you eventually migrate this to 1 -> 2 (while you make sure the equivalent information is always available)?

Reply:
>What about we just keep the current implementation and you eventually migrate this to 1 -> 2 (while you make sure the equivalent information is always available)?

@rkooo567 That sounds like the best route to me!

>Can you tell me what it means by it gets delayed anyway?

I believe he's talking about the possibility of that dummy handler being queued before actually executing on the event loop, and that queueing of that handler not blocking the submission of the underlying RPC. In fact, since this handler is posted after the `.StartCall()` call, there's a chance that the RPC will be submitted before this line is even reached.

Reply:
I see, that makes sense. Thanks for the response :)! 

@ericl do you think the suggested path sounds good to you? I am also okay to just remove it given probably the same information will be available soon. Having this in the master will help me validating all of my fixes, so I prefer to have it though.

Reply:
How about we remove it then? @clarkzinzow looking forwards to having general purpose Start/End op tracking :)

Reply:
That also works. I will just pick it up for my working branch then.

Reply:
@ericl Yep ETA is this week! 😁

Reply:
This is removed.  @clarkzinzow can you assign me as a reviewer for the PR? (also, please include the grpc request/reply stuff there!)